lib/dmv.rb

```diff
@@ -9,8 +13,166 @@ def add_facility(facility)
   end
 
   def facilities_offering_service(service)
-    @facilities.find do |facility|
+    @facilities.find_all do |facility|
       facility.services.include?(service)
     end
   end
+
+  #Larger methods below.  For future: could consider classes for DMVs in each state, to better handle
+  #idiosyncracies to procedures and different dataset formats for different states.
+
+  def create_state_facilities(state, facilities_incoming_data)
+    facilities_incoming_data.each do |facility_data|
+      #Construct facility based on state, since dataset must be converted differently due to idiosyncracies in each case
+      if state == "Colorado"
+        new_facility = Facility.new(facility_parameters_co(facility_data))
+      elsif state == "New York"
+        new_facility = Facility.new(facility_parameters_ny(facility_data))
+      elsif state == "Missouri"
+        new_facility = Facility.new(facility_parameters_mo(facility_data))
+      end
+
+      add_facility(new_facility)
+
+      #Add services.  CO specifies some of them, so those must be checked (in furture rewrite, could remake initialize() to deal with this more cleanly too)
+      new_facility.add_service("Written Test")
+      new_facility.add_service("Road Test")
+      if state == "Colorado"
+        new_facility.add_service("New Drivers License") if facility_data[:services_p].include?("registration")
+        new_facility.add_service("Renew Drivers License") if facility_data[:services_p].include?("renew")
+      else
+        new_facility.add_service("New Drivers License")
+        new_facility.add_service("Renew Drivers License")
+      end
+    end
+  end
+
+  def get_ev_registration_analytics(state, specified_year)
+    #Determine most popular make/model registered, the # registered for specified year, and the county with most registered vehicles
+    #NOTE: I kept method naming based on EV vehicles.  However, now that I've extended functionality to allow NY vehicles (later in iteration 4), this method is more general,
+    #and can analyze general NY vehicles.
+
+    vehicle_tally = {}
```

Comment:
Consider breaking up some of this logic into well-named helper methods in order to shorten this method. We typically want methods to stay between 8-10 lines, so using a helper method like tally_vehicle could be a great way to extract some of this into a separate method.